# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1205292827 14400
#      Tue Mar 11 23:33:47 2008 -0400
# Branch release-3-0-x
# Node ID 842cc9439011872cc7be43883030a3095e10a1ca
# Parent  e13219eaf388da734744b16d1253d340a60e6751
graphcs.cc, graphics.h.in: new property, linewidth

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,12 @@
 2008-03-11  John W. Eaton  <jwe@octave.org>
 
+	* graphics.cc, graphics.h.in (class axes): New property, linewidth.
+
 	* DLD-FUNCTIONS/eig.cc (Feig): Handle possible error from EIG.
 	* DLD-FUNCTIONS/qp.cc (qp, Fqp): Likewise.
 	* xpow.cc (xpow): Likewise.
 
 2008-03-10  John W. Eaton  <jwe@octave.org>
 
 	* mex.cc (mxCreateLogicalScalar): Argument is now mxLogical.
 
diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -1135,16 +1135,17 @@ axes::properties::properties (const grap
     xscale (radio_values ("{linear}|log")),
     yscale (radio_values ("{linear}|log")),
     zscale (radio_values ("{linear}|log")),
     xdir ("normal"),
     ydir ("normal"),
     zdir ("normal"),
     xaxislocation ("bottom"),
     yaxislocation ("left"),
+    linewidth (0.5),
     view (),
     visible ("on"),
     nextplot ("replace"),
     outerposition (),
     activepositionproperty (radio_values ("{outerposition}|position")),
     __colorbar__ (radio_values ("{none}|north|south|east|west|northoutside|southoutside|eastoutside|westoutside"))
 {
   Matrix tlim (1, 2, 0.0);
@@ -1343,16 +1344,18 @@ axes::properties::set (const caseless_st
   else if (name.compare ("ydir"))
     set_ydir (val);
   else if (name.compare ("zdir"))
     set_zdir (val);
   else if (name.compare ("xaxislocation"))
     set_xaxislocation (val);
   else if (name.compare ("yaxislocation"))
     set_yaxislocation (val);
+  else if (name.compare ("linwdiwth"))
+    set_linewidth (val);
   else if (name.compare ("view"))
     set_view (val);
   else if (name.compare ("visible"))
     set_visible (val);
   else if (name.compare ("nextplot"))
     set_nextplot (val);
   else if (name.compare ("outerposition"))
     set_outerposition (val);
@@ -1427,16 +1430,17 @@ axes::properties::set_defaults (base_gra
   xscale = radio_property (radio_values ("{linear}|log"));
   yscale = radio_property (radio_values ("{linear}|log"));
   zscale = radio_property (radio_values ("{linear}|log"));
   xdir = "normal";
   ydir = "normal";
   zdir = "normal";
   xaxislocation = "left";
   yaxislocation = "bottom";
+  linewidth = 0.5;
 
   Matrix tview (1, 2, 0.0);
   tview(1) = 90;
   view = tview;
 
   visible = "on";
   nextplot = "replace";
 
@@ -1553,16 +1557,17 @@ axes::properties::get (void) const
   m.assign ("xscale", xscale);
   m.assign ("yscale", yscale);
   m.assign ("zscale", zscale);
   m.assign ("xdir", xdir);
   m.assign ("ydir", ydir);
   m.assign ("zdir", zdir);
   m.assign ("xaxislocation", xaxislocation);
   m.assign ("yaxislocation", yaxislocation);
+  m.assign ("linewidth", linewidth);
   m.assign ("view", view);
   m.assign ("visible", visible);
   m.assign ("nextplot", nextplot);
   m.assign ("outerposition", outerposition);
   m.assign ("activepositionproperty", activepositionproperty);
   m.assign ("__colorbar__", __colorbar__);
 
   return m;
@@ -1680,16 +1685,18 @@ axes::properties::get (const caseless_st
   else if (name.compare ("ydir"))
     retval = ydir;
   else if (name.compare ("zdir"))
     retval = zdir;
   else if (name.compare ("xaxislocation"))
     retval = xaxislocation;
   else if (name.compare ("yaxislocation"))
     retval = yaxislocation;
+  else if (name.compare ("linewidth"))
+    retval = linewidth;
   else if (name.compare ("view"))
     retval = view;
   else if (name.compare ("visible"))
     retval = visible;
   else if (name.compare ("nextplot"))
     retval = nextplot;
   else if (name.compare ("outerposition"))
     retval = outerposition;
@@ -1789,16 +1796,17 @@ axes::properties::factory_defaults (void
   m["xscale"] = radio_property (radio_values ("{linear}|log"));
   m["yscale"] = radio_property (radio_values ("{linear}|log"));
   m["zscale"] = radio_property (radio_values ("{linear}|log"));
   m["xdir"] = "normal";
   m["ydir"] = "normal";
   m["zdir"] = "normal";
   m["xaxislocation"] = "bottom";
   m["yaxislocation"] = "left";
+  m["linewidth"] = 0.5;
 
   Matrix tview (1, 2, 0.0);
   tview(1) = 90;
 
   m["view"] = tview;
 
   m["visible"] = "on";
   m["nextplot"] = "replace";
diff --git a/src/graphics.h.in b/src/graphics.h.in
--- a/src/graphics.h.in
+++ b/src/graphics.h.in
@@ -1409,16 +1409,17 @@ public:
       radio_property xscale al
       radio_property yscale al
       radio_property zscale al
       octave_value xdir
       octave_value ydir
       octave_value zdir
       octave_value xaxislocation
       octave_value yaxislocation
+      octave_value linewidth
       octave_value view
       octave_value visible
       octave_value nextplot
       octave_value outerposition
       radio_property activepositionproperty a
       radio_property __colorbar__ a
    END_PROPERTIES
 
